### PR TITLE
Add support for simulating autoupdates locally.

### PR DESCRIPTION
**Pull request by xiam**
This is the client part required to [stage autoupdates](https://github.com/getlantern/autoupdate-server/pull/2).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -103,6 +103,8 @@ LANTERN_MOBILE_ANDROID_RELEASE := $(LANTERN_MOBILE_DIR)/app/build/outputs/apk/ap
 LANTERN_YAML := lantern.yaml
 LANTERN_YAML_PATH := installer-resources/lantern.yaml
 
+BUILD_TAGS ?=
```
Comment (xiam) [inline Makefile]: An example would be

```
BUILD_TAGS=mockupdate make linux-amd64
```

**Comment by myleshorton** [comment]
Nice! I actually also think we should allow betas to be updated from but
not to.

**Comment by xiam** [comment]
> Nice! I actually also think we should allow betas to be updated from but not to

I think this is something that can be solved by not releasing the `update_*` blob for betas, right?

**Comment by xiam** [comment]
The corresponding part on the update-server is here: https://github.com/getlantern/autoupdate-server/pull/2

**Comment by myleshorton** [comment]
LGTM @xiam. Out of curiosity, how would actually hitting the mock update server versus the live server work? Not related to this merge though.

File: src/github.com/getlantern/flashlight/app/mock_autoupdate.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// +build mockupdate
+
+package app
+
+// This is a public key derived from a private key that is freely available.
+// Don't use in production.
+const packagePublicKey = `-----BEGIN PUBLIC KEY-----
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/app/mock_autoupdate.go]: I'm a bit confused here @xiam . `packagePublicKey` is used nowhere , while `src/github.com/getlantern/autoupdate/keys.go` contains `PackagePublicKey` which is the same as what defined in  `src/github.com/getlantern/flashlight/app/autoupdate.go`.

Comment (xiam) [inline src/github.com/getlantern/flashlight/app/mock_autoupdate.go]: Hmm, we do use `packagePublicKey` in different places:

```
grep -R packagePublicKey .
./github.com/getlantern/flashlight/app/mock_autoupdate.go:const packagePublicKey = `-----BEGIN PUBLIC KEY-----
./github.com/getlantern/flashlight/app/app.go:  autoupdate.PublicKey = []byte(packagePublicKey)
./github.com/getlantern/flashlight/app/autoupdate.go:const packagePublicKey = `-----BEGIN PUBLIC KEY-----
```

And, keys are also different:

```
vagrant@lantern:~/go/src/github.com/getlantern/lantern-devel/src$ cat ./github.com/getlantern/flashlight/app/mock_autoupdate.go
// +build mockupdate

package app

// This is a public key derived from a private key that is freely available.
// Don't use in production.
const packagePublicKey = `-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzSoibtACnqcp2uTGjCMJ
tTOLDIMQ4oGPhGHT4Q/epum+H3hcbBNs9jRnMRWgX4z++xxuNJnhmoJw0eUXB7B4
vj5DYpPajq6gPY8JuraF4ngfP5oxKj2BqpEUR9bx+3SjOSInrirM0JZO+aAW38BQ
NJB+sS7JvbPjcwdjwKc5IKzc9kxxJNoZoFE9GMnYzaOrAlpCuAKWH8SCXYtCTxsX
fKexdDxsI5Vzm5lQHJLMeqhLTQTUm9oQofwNAOGOkn6dD4ObMlmFTOsf1G03/Dl9
sVgjaWaZ9bGjvJ9B85UxNeWwduy+uMrqFytxG6bbq0PbDEVu6ZQCPyiyCA7l945J
OQIDAQAB
-----END PUBLIC KEY-----`
vagrant@lantern:~/go/src/github.com/getlantern/lantern-devel/src$ cat ./github.com/getlantern/flashlight/app/autoupdate.go
// +build !mockupdate

package app

// This is the public key of the BNS cert. Incoming updates will be signed to
// prevent MITM attacks.
const packagePublicKey = `-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxeReZ0VHDQ+/XYEHhFq0
krT+a/+/mlhCkgJ/605KmPXqBv8qo5f1iK6C+TQ87264J4Z9yw0tRwcdY1/ofpH7
Tywq3pBOgfrnnP9gFtquQ/tgzVkorQ0L51w9HLZ3cCjpaLpofIaztgbCIzsCT6kV
Nx6Sd/4KBSuThhMEnP5pu5Wxr4/lujIpTeVEXzljQMxqX+58ISeXYx6SxLXx5Vgj
1IB6NJwjg7r4Nzg/zUH0ZkCWj3rDWo6itIoeo61o+hPQAjH23TCKOn8Ssaejocyg
CrcOc7aqfGuVM3HuHxtXsjYPqJMVHiXKosi9HcHo5BACPT0FkrZIwz3k6Vy1h7nB
HQIDAQAB
-----END PUBLIC KEY-----`
```

Comment (fffw) [inline src/github.com/getlantern/flashlight/app/mock_autoupdate.go]: Which branch were you grepping @xiam ? I couldn't find below line on `pro-backend` branch.

./github.com/getlantern/flashlight/app/app.go:  autoupdate.PublicKey = []byte(packagePublicKey)

Comment (xiam) [inline src/github.com/getlantern/flashlight/app/mock_autoupdate.go]: That's probably the cause, this PR was merged into `devel`. Not sure is this is already on `pro-backend`.

Comment (fffw) [inline src/github.com/getlantern/flashlight/app/mock_autoupdate.go]: Oh, I see. The line is removed in `pro-backend`

Comment (xiam) [inline src/github.com/getlantern/flashlight/app/mock_autoupdate.go]: I opened an issue to find out why https://github.com/getlantern/lantern-pro/issues/119, this should be working on devel, tho.

**Pull request merged**